File: backend/internal/models/meet.go
package models

import (
	"time"
)

// Meet — мероприятие
type Meet struct {
	ID           int     `gorm:"primaryKey;autoIncrement"`
	EventName    *string `gorm:"type:text"`
	CustomerName *string `gorm:"type:text"`
	Email        *string `gorm:"type:text"`
	Phone        *string `gorm:"type:text"`
	Location     *string `gorm:"type:text"`
	Platform     *string `gorm:"type:text"`
	Devices      *string `gorm:"type:text"`
	URL          *string `gorm:"type:text"`
	ShortURL     *string `gorm:"type:text"`

	Status      string  `gorm:"type:text;default:'new'"`
	Description *string `gorm:"type:text"`

	Admin *string `gorm:"type:text;"`

	Start     *time.Time
	End       *time.Time
	CreatedAt time.Time  `gorm:"autoCreateTime"`
	UpdatedAt *time.Time `gorm:"autoUpdateTime"`
}

File: frontend/src/interfaces/response/meet.ts
import type { Pagination } from "./pagination";

export interface MeetResponse {
  id: number;
  eventName: string;
  customerName: string;
  email: string;
  phone: string;
  location: string;
  platform: string;
  devices: string;

  url?: string | null; // может быть null или отсутствовать
  shortUrl?: string | null; // может быть null или отсутствовать
  status: string;
  description?: string | null; // может быть null или отсутствовать
  admin?: string | null; // может быть null или отсутствовать

  start: string; // ISO дата-время
  end: string; // ISO дата-время
  CreatedAt: string; // дата создания
  UpdatedAt: string; // дата обновления
}

export interface MeetsListResponse {
  data: MeetResponse[]; // массив встреч, может быть пустым []
  pagination: Pagination;
}
